internal/resolution/client/depsdev_client.go

```diff
@@ -0,0 +1,96 @@
+package client
+
+import (
+	"context"
+	"encoding/gob"
+	"os"
+
+	pb "deps.dev/api/v3alpha"
+	"deps.dev/util/resolve"
+	"github.com/google/osv-scanner/internal/resolution/datasource"
+)
+
+const depsDevCacheExt = ".resolve.deps"
+
+// DepsDevClient is a ResolutionClient wrapping the official resolve.APIClient
+type DepsDevClient struct {
+	resolve.APIClient
+	c *datasource.DepsDevAPIClient
+}
+
+func NewDepsDevClient(addr string) (*DepsDevClient, error) {
+	c, err := datasource.NewDepsDevAPIClient(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DepsDevClient{APIClient: *resolve.NewAPIClient(c), c: c}, nil
+}
+
+func (d *DepsDevClient) PreFetch(ctx context.Context, requirements []resolve.RequirementVersion, manifestPath string) {
+	//nolint:errcheck // It doesn't matter if loading the cache fails
+	d.LoadCache(manifestPath)
+
+	// Use the deps.dev client to fetch complete dependency graphs of the direct requirements
+	for _, im := range requirements {
+		// Get the preferred version of the import requirement
+		vks, err := d.MatchingVersions(ctx, im.VersionKey)
+		if err != nil || len(vks) == 0 {
+			continue
+		}
+
+		vk := vks[len(vks)-1]
+
+		// Make a request for the precomputed dependency tree
+		resp, err := d.c.GetDependencies(ctx, &pb.GetDependenciesRequest{
+			VersionKey: &pb.VersionKey{
+				System:  pb.System(vk.System),
+				Name:    vk.Name,
+				Version: vk.Version,
+			},
+		})
+		if err != nil {
+			continue
+		}
+
+		// Send off queries to cache the packages in the dependency tree
```

Comment:
Is this caching remote (on deps.dev servers) or local (guessing from datasource/depsdev_api.go)?  Trying to figure out how this caches packages.

Reply:
This is local - The [deps.dev implementation of the client](https://github.com/google/deps.dev/blob/main/util/resolve/api.go) calls the relevant `datasource/depsdeb_api.go` methods, which will fill in its own maps.